code/modules/modular_computers/file_system/programs/system/file_browser.dm

```diff
@@ -28,28 +39,48 @@
 		var/obj/item/weapon/computer_hardware/hard_drive/HDD = computer.hard_drive
 		if(!HDD)
 			return 1
+		var/newpassword = sanitize(input(usr, "Enter a password or leave blank to leave unprotected:"))
 		var/datum/computer_file/data/F = new/datum/computer_file/data()
 		F.filename = newname
 		F.filetype = "TXT"
+		F.password = newpassword
 		HDD.store_file(F)
 	if(href_list["PRG_deletefile"])
 		. = 1
 		var/obj/item/weapon/computer_hardware/hard_drive/HDD = computer.hard_drive
 		if(!HDD)
 			return 1
 		var/datum/computer_file/file = HDD.find_file_by_name(href_list["PRG_deletefile"])
+		var/checkpass = ""
+		var/currentpass = file.password
 		if(!file || file.undeletable)
 			return 1
-		HDD.remove_file(file)
+		if(!currentpass)
+			HDD.remove_file(file)
+		if(currentpass)
+			checkpass = sanitize(input(usr, "Please enter the password:"))
+			if(checkpass == currentpass)
+				HDD.remove_file(file)
+			if(!checkpass == currentpass)
```

Comment:
This can be an else statement, and I think it's written wrong in the first place.
NOT EQUALS statements are `x != y` not `!x == y', I think I'm seeing this a lot in this PR so you might want to look at that.